tests/element_test_vectors.rs

```diff
@@ -67,13 +63,15 @@ fn serialize(format: Format, elements: &[Element]) -> IonResult<Vec<u8>> {
 }
 
 trait ElementApi {
+    type ElementReader<'a>: ElementReader;
```

Comment:
🗺️ I loosened the integration tests from making a `Reader` to making something that implements `ElementReader` so I could add a test suite for the `LazyReader`. However, I ran into another issue: the tests require whatever reader you create to also be able to read text. I considered making a `Box<dyn ElementReader>` that could be either a `LazyReader` (binary) or a `Reader` (binary/text) when the binary IVM is missing. However, the current definition of the `ElementReader` trait is not object safe, preventing it from being used with `dyn`. 😞 All of this is tractable, but will take some more refactoring effort.

I have punted on the integration tests for the moment, but will add them in a follow-on PR. I believe this is ok for a gated experimental feature, but could be convinced to add that to this (already enormous) PR if there's call for it.

Reply:
Opened #554.

> is what I did in https://github.com/amazon-ion/ion-rust/pull/542 equivalent?

No, since that still requires an implementation of `IonReader` which (as you've experienced) can be a pain to work with. I was trying to dumb it down to taking anything that could produce `Vec<Element>` (i.e. `ElementReader` impls).